### DISCUSSION TITLE
use-form "A component is changing an uncontrolled input to be controlled" ... Does it matter?

### DISCUSSION THREAD

**Discussion by juliuslissau** [mantinedev/mantine#1860] [Feedback]
I just noticed this in the v5 release. Possibly it was there before:

If initialValues isn't set for all fields in useForm then React will give the error/warning "A component is changing an uncontrolled input to be controlled..."

This makes sense, but does it actually matter? Could I ignore the warning and just leave initialValues as an empty object? It seems like my website works just fine either way.

**Comment by rtivital** [comment]
I advise to always have initial values that match your final data. If it works for you without the initial values, then you can ignore the warning.

**Reply by juliuslissau** [reply]
Alright thanks!

**Comment by kevinsmouts** [comment]
@rtivital I'm in a situation where the step by step (Mantine Stepper) validation is failing on all steps because of values initialized as empty.
To my knowledge there is no method to form.validate() only the inputs present in the current step (other than hardcode it). So the only way I've found is to not define the initialValues, but then I get this React error. A component is changing an uncontrolled input to be controlled. 

Is there a way with useForm to make a specific input controlled even though its temporary value is undefined?

**Reply by rtivital** [reply]
https://mantine.dev/form/recipes/#form-with-multiple-steps

**Reply by kevinsmouts** [reply]
wow, thanks. can't believe I missed that given the time I spent on that doc over the past few days. it could be worth bringing this example up in the form validation section.
